executor/src/witgen/machines/machine_extractor.rs

```diff
@@ -216,8 +216,16 @@ impl<'a, T: FieldElement> MachineExtractor<'a, T> {
         // receiving end of a lookup, we remove any multiplicity columns here.
         // Note that we don't use the passed identities, because we want to
         // include removed range constraints.
-        let multiplicity_columns = convert(&self.fixed.analyzed.identities)
+        let mut dummy_id_counter = 1..;
+        let multiplicity_columns = self
+            .fixed
+            .analyzed
+            .identities
             .iter()
+            // We're only interested in bus receives, and polynomial identities
+            // for sure don't translate to bus receives, so we can filter them out.
+            .filter(|identity| !matches!(identity, AnalyzedIdentity::Polynomial(_)))
+            .flat_map(|identity| convert_identity(&mut dummy_id_counter, identity))
```

Comment:
Is there really no better way than re-doing the conversion all over again here just to get the multiplicity columns?